internal/operands/tekton-tasks/tekton-tasks.go

```diff
@@ -78,76 +76,74 @@ type tektonTasks struct {
 
 var _ operands.Operand = &tektonTasks{}
 
-func New(bundle *tektonbundle.Bundle) *tektonTasks {
-	tt := &tektonTasks{
+func New(bundle *tektonbundle.Bundle) operands.Operand {
+	newTasks := []pipeline.Task{}
+	for _, task := range bundle.Tasks {
+		if _, ok := AllowedTasks[task.Name]; ok {
+			newTasks = append(newTasks, task)
+		}
+	}
+	bundle.Tasks = newTasks
```

Comment:
**future PR** - IMHO It's not a good design to modify the `budnle` parameter.
I would not expect a function called `New()` to modify any of it's parameters.